server.go

```diff
@@ -84,6 +84,7 @@ Loop:
 					echoerr(c, "listen: conn: client id should be a number")
 				} else {
 					gConnList[id] = c
+					return
```

Comment:
No, it's not a separate bugfix, it's required to make the whole `recv` mechanism work:

1. The main `lf` instance starts, which sends a `conn` message to the server to register itself.
2. In response, the server creates a thread (A) to handle the connection from the main instance.
3. Some time later, `lf -remote` is called to query for some information. This is sent to the server, which will in turn forward the query to the main instance.
4. In response the server creates another thread (B) to handle the connection from the remote instance.
5. The main instance writes the requested information back to the server. However the problem is that there are two server threads (A and B) now reading from the connection to the main instance, and they conflict with each other.

Only thread B should read the information since it is connected to the remote instance and needs to relay the information back to it. In fact after thread A handles the `conn` message, there is no need for it to do anything else, hence the `return`.

Reply:
This is complicated, I'm still trying to understand it and am wondering how it could be explained in comments or code.

Just to double-check that I'm on the right track, do I understand correctly that with this `return` nobody is going to be listenting to this connection. So, the correctness of the design depends on the fact that an `lf` client never sends anything to the server after the first `conn` request. OTOH, the server can still send messages to the client via `gConnList[id]`.

Reply:
I'm not sure how to best phrase it as a comment because it already makes sense to me, but to answer your questions:

> Just to double-check that I'm on the right track, do I understand correctly that after this `return`, the server is not going to be listening to this connection at all?

The server thread that handles the initial `conn` request `return`s immediately, doesn't do anything else and terminates. From then on the connection is stored in a map and can be read from and written to by other server threads (e.g. created to handle `lf -remote ...`), until a `drop` request deletes the connection from the map.

> So, the correctness of the design depends on the fact that an `lf` client never sends anything to the server after the first `conn` request

Yes that's correct, the `readExpr` thread initiates the connection and then just handles commands sent from the server. Previously this was just executing `send` commands, but now this includes handling `query` commands as well. Any commands that the client needs to send to the server (e.g. `load` in various places in the code) is actually done by creating a new connection, and from the server's point of view is indistinguishable from if you had just typed `lf -remote "send load"` on the command line.

Reply:
I might add a comment something the lines of:

> The connection is now managed by the server and will be cleaned up when receiving a `drop` command.